### PR TITLE
Add new block failure reasons

### LINKED ISSUE #1466: Add new BlockFailure variants
https://github.com/iotaledger/iota.go/pull/572/files

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Adds new block failure reasons from iota.go.

## Links to any relevant issues

Closes #1466

### CODE REVIEW

File: sdk/src/types/api/core/response.rs
Comment (thibault-martinez) [inline sdk/src/types/api/core/response.rs:1]: Do we have this enum in nodejs/python as well?

Comment (DaughterOfMars) [reply sdk/src/types/api/core/response.rs:1]: Apparently not in nodejs, but it is in python

Comment (DaughterOfMars) [reply sdk/src/types/api/core/response.rs:1]: I added it to both

File: bindings/nodejs/lib/types/models/block-metadata.ts [hunk 1]
```diff
@@ -13,35 +15,19 @@ export interface IBlockMetadata {
...
+     * The block state.
      */
-    referencedByMilestoneIndex?: number;
+    blockState?: BlockState;
     /**
-     * Is this block a valid milestone.
+     * The transaction state.
      */
-    milestoneIndex?: number;
+    txState?: TransactionState;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/models/block-metadata.ts]: This is `transactionState` in the TIP

File: bindings/nodejs/lib/types/models/block-metadata.ts [hunk 2]
```diff
@@ -13,35 +15,19 @@ export interface IBlockMetadata {
...
-    transactionFailureReason?: TransactionFailureReason;
-    /**
-     * Should the block be promoted.
-     */
-    shouldPromote?: boolean;
-    /**
-     * Should the block be reattached.
-     */
-    shouldReattach?: boolean;
+    txFailureReason?: TransactionFailureReason;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/models/block-metadata.ts]: This is `transactionFailureReason` in the TIP

File: bindings/nodejs/lib/types/models/block-metadata.ts (hunk 1 above)
Comment (thibault-martinez) [reply bindings/nodejs/lib/types/models/block-metadata.ts]: Asked on Slack. If ever it changed, needs to be applied to Python as well.

File: bindings/nodejs/lib/types/models/block-metadata.ts [hunk 3]
```diff
@@ -13,35 +15,19 @@ export interface IBlockMetadata {
...
-    /**
-     * Is the block solid.
-     */
-    isSolid: boolean;
-    /**
-     * Is the block referenced by a milestone.
+     * The block state.
      */
-    referencedByMilestoneIndex?: number;
+    blockState?: BlockState;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/models/block-metadata.ts]:
```suggestion
    blockState: BlockState;
```
Not optional. Can you do the same in Python and remove the TODO please ?

Comment (DaughterOfMars) [reply bindings/nodejs/lib/types/models/block-metadata.ts]: sure

**Review by thibault-martinez — APPROVED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Conflicts

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**